docs/faq/troubleshooting.txt

```diff
@@ -34,6 +34,42 @@ To do this, open Terminal.app and navigate (using the ``cd`` command) to the
 directory where :doc:`django-admin </ref/django-admin>` is installed, then
 run the command ``sudo chmod +x django-admin``.
 
+.. _troubleshooting-external-dependencies:
+
+External Dependencies
+=============================
+
+Installing dependencies fails
+----------------------------------------
+
+Sometimes, when you want to implement extra functionality to your project
+you are probably going to install some extra applications. However these 
+programs usually rely on other programs or specific external libraries, called 
+dependencies.
+
+When installing dependencies you might run across similar errors like the 
+following one:
+
+.. code-block:: console
+
+      ...
+    $ Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-PCA0QU/argon2-cffi/
+
+On the other hand dependency is built using a few other libraries. 
+Lacking any of the latter may cause you problems like the one above.
+
+The best way to deal with such errors, when installing dependencies,
+is to check the dependency's documentation in search for required libraries.
+
+In our case we were missing *libffi* library used in 
+`argon2-cffi <https://argon2-cffi.readthedocs.io/en/stable/installation.html>`_.
+All you need to do is install the missing library using 
+``sudo apt-get install libffi-dev``.
```

Comment:
@timgraham Learning everyday.

What you wrote is absolutely true. Will do as suggested.